### PR TITLE
intel-gpu-resource-driver: update to v0.5.1 version

### PR DESCRIPTION
**Pull request by oxxenix**
The helm chart is updated with new images, CRDs. The template file was modified to remove unnecessary settings. The documentation includes more information about the project.

### CODE REVIEW

**Comment by oxxenix** [comment]
@byako, @eero-t, PTAL

File: charts/intel-gpu-resource-driver/templates/clusterrole.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
-{{- if and .Values.rbac.create }}
+{{- if .Values.rbac.create }}
```
Comment (byako) [inline charts/intel-gpu-resource-driver/templates/clusterrole.yaml:1]: Same question as in https://github.com/intel/helm-charts/pull/57#discussion_r1691396061 : does the resource driver work without the RBAC? What will happen when user inevitably selects not to install RBAC rules? Why are we giving this option ?

**Review by byako — COMMENTED** [review]
LGTM, but I'm worried about the RBAC being optional. See the comment.

File: charts/intel-gpu-resource-driver/values.yaml [hunk 2]
```diff
@@ -37,45 +37,10 @@ controller:
     {}
     #node-role.kubernetes.io/control-plane: ""
   affinity: {}
-  containers:
```
Comment (byako) [inline charts/intel-gpu-resource-driver/values.yaml:42]: Line 20 in this file above still has `rbac` section making an impression that it is used. Is it used?

**Review by byako — CHANGES_REQUESTED** [review]
vlues.yaml has rbac sectino still with `create: true` and other useless fields.

File: charts/intel-gpu-resource-driver/values.yaml (hunk 2 above)
Comment (oxxenix) [reply charts/intel-gpu-resource-driver/values.yaml:42]: that was overlooked. now I removed that completely

**Review by byako — APPROVED** [review]

File: charts/intel-gpu-resource-driver/README.md [hunk 3]
```diff
@@ -1,5 +1,12 @@
 # Dynamic Resource Allocation (DRA) Intel GPU Driver Helm Chart
 
+## The chart installs GPU resource driver:
+
+- [GPU](https://github.com/intel-innersource/containers.orchestrators.kubernetes.intel-resource-drivers-for-kubernetes/doc/gpu/README.md)
```
Comment (byako) [inline charts/intel-gpu-resource-driver/README.md:5]: Sorry, just noticed this, please replace with the public links.

**Pull request merged**